shared/constants/types/fs.js

```diff
@@ -2,7 +2,27 @@
 import * as I from 'immutable'
 
 export type _State = {
-  counter: number,
+  path: Path,
+  pathItems: I.Map<Path, PathItem>,
 }
 
 export type State = I.RecordOf<_State>
+
+export type PathItem = {
+  type: PathType,
+  children: I.List<string> | null,
```

Comment:
you want a disjoint union here and stronger typing. aka i *think* the intention is something like
```
type _PathItem = 
| {
  type: 'file'
}  
| {
type: 'folder' | 'symlink' | 'exec',
children: I.List<Path>
}
```

Reply:
Yeah I thought of doing that but wasn't sure how much to trust Flow (given its failure to discover that my `FolderProps` were getting an Immutable object instead of an array).